libs/payments/stripe/src/lib/stripe.error.ts

```diff
@@ -43,40 +41,23 @@ export class ProductNotFoundError extends StripeError {
 }
 
 export class PromotionCodeCouldNotBeAttachedError extends StripeError {
-  constructor(cause: Error) {
-    super('Promotion code could not be attached to subscription', cause);
-  }
-}
-
-export class PromotionCodeInvalidError extends StripeError {
-  constructor() {
-    super('Invalid promotion code');
-  }
-}
-
-export class PromotionCodeNotForSubscriptionError extends StripeError {
-  constructor() {
-    super(
-      "Promotion code restricted to a product that doesn't match the product on this subscription"
-    );
-  }
-}
+  customerId: string | undefined;
+  subscriptionId: string | undefined;
+  promotionId: string | undefined;
 
-export class SubscriptionCustomerIdDoesNotMatchCustomerIdError extends StripeError {
-  constructor() {
-    super('subscription.customerId does not match passed in customerId');
-  }
-}
-
-export class SubscriptionNotActiveError extends StripeError {
-  constructor() {
-    super('Subscription is not active');
-  }
-}
-
-export class SubscriptionPriceUnknownError extends StripeError {
-  constructor() {
-    super('Unknown subscription price');
+  constructor(
+    message: string,
+    data?: {
+      customerId?: string;
+      subscriptionId?: string;
+      promotionId?: string;
+    },
+    cause?: Error
```

Comment:
Cause should likely go first since it'll be passed along to `super` in that order.